testing/item-renderer-hooks.tsx

```diff
@@ -0,0 +1,190 @@
+            showSolutions,
+        ),
+    );
+
+    // Get a derived item based on answerless state
+    const renderedItem = React.useMemo(() => {
+        const shouldUseAnswerless =
+            state.answerless &&
+            !state.reviewMode &&
+            (state.showSolutions === "none" || !state.showSolutions);
+
+        return shouldUseAnswerless ? splitPerseusItem(item) : item;
+    }, [item, state.answerless, state.reviewMode, state.showSolutions]);
+
+    // Update item if the renderedItem changes
+    useEffect(() => {
+        dispatch({type: "UPDATE_ITEM", payload: renderedItem});
+    }, [renderedItem]);
+
+    // Show the popover when the score changes
+    useEffect(() => {
+        if (!state.score) {
+            return;
+        }
+
+        // If the answer is not empty or there's a message, show the popover
+        if (!state.score.empty || state.score.message !== null) {
+            // This is a hack to ensure the popover is shown after the score is set
+            // so that it gets the correct position.
+            setTimeout(() => {
+                dispatch({type: "TOGGLE_POPOVER", payload: true});
+            }, 100);
+        }
+    }, [state.score]);
+
+    // Create API options
+    const options = React.useMemo(
+        () => ({
+            ...apiOptions,
+            isMobile: state.isMobile,
+            customKeypad: state.isMobile, // Use the mobile keypad for mobile
+            showSolutions: undefined,
+            interactionCallback: () => {
+                if (state.showPopover) {
+                    dispatch({type: "TOGGLE_POPOVER", payload: false});
+                    // Only deselect incorrect choices when the score is not empty
+                    // This prevents deselection when clicking a new answer after
+                    // receiving an empty score
+                    if (state.score && !state.score.empty) {
+                        ref.current?.deselectIncorrectSelectedChoices();
+                    }
+                }
+            },
+        }),
+        [apiOptions, state.isMobile, state.showPopover, state.score],
+    );
+
+    /**
+     * Get the score for the current user input
+     */
+    const getScore = React.useCallback((): KEScore | undefined => {
+        const renderer = ref.current;
+        if (!renderer) {
+            return undefined;
+        }
+
+        const userInput = renderer.getUserInput();
+        const score = scorePerseusItem(
+            state.perseusItem.question,
+            userInput,
+            "en",
+        );
+
+        // Continue to include an empty guess for the now defunct answer area.
+        const maxCompatGuess = [renderer.getUserInputLegacy(), []];
+
+        const keScore = keScoreFromPerseusScore(
+            score,
+            maxCompatGuess,
+            renderer.getSerializedState().question,
+        );
+
+        if (!keScore.empty) {
+            ref.current?.showRationalesForCurrentlySelectedChoices();
```

Comment:
Note that this will be removed when we remove the rationales (See https://github.com/Khan/perseus/pull/2540)